src/Categories/Category/Monoidal/Interchange.agda

```diff
@@ -0,0 +1,603 @@
+
+  -- An alternative definition of the interchange that associates differently.
+
+  swapInner′ : ∀ {W X Y Z} → (W ⊗₀ X) ⊗₀ (Y ⊗₀ Z) ≅ (W ⊗₀ Y) ⊗₀ (X ⊗₀ Z)
+  swapInner′ = α ∘ᵢ (α ⁻¹ ∘ᵢ idᵢ ⊗ᵢ σ ∘ᵢ α) ⊗ᵢ idᵢ ∘ᵢ α ⁻¹
+
+  module swapInner′ {W X Y Z} = _≅_ (swapInner′ {W} {X} {Y} {Z})
+  private
+    i′⇒ = swapInner′.from
+    i′⇐ = swapInner′.to
+
+  -- Derived coherence laws.
+
+  swapInner-coherent : ∀ {W X Y Z} → i⇒ ≈ i′⇒ {W} {X} {Y} {Z}
+  swapInner-coherent = begin
+      α⇐ ∘ id ⊗₁ (α⇒ ∘ σ⇒ ⊗₁ id ∘ α⇐) ∘ α⇒
+    ≈⟨ refl⟩∘⟨ pushˡ split₂ˡ ⟩
+      α⇐ ∘ id ⊗₁ α⇒ ∘ id ⊗₁ (σ⇒ ⊗₁ id ∘ α⇐) ∘ α⇒
+    ≈⟨ refl⟩∘⟨ refl⟩∘⟨ pushˡ split₂ˡ ⟩
+      α⇐ ∘ id ⊗₁ α⇒ ∘ id ⊗₁ (σ⇒ ⊗₁ id) ∘ id ⊗₁ α⇐ ∘ α⇒
+    ≈⟨ refl⟩∘⟨ refl⟩∘⟨ refl⟩∘⟨ conjugate-from α (idᵢ ⊗ᵢ α) (⟺ pentagon) ⟩
+      α⇐ ∘ id ⊗₁ α⇒ ∘ id ⊗₁ (σ⇒ ⊗₁ id) ∘ (α⇒ ∘ α⇒ ⊗₁ id) ∘ α⇐
+    ≈˘⟨ pullʳ (pullʳ (extendʳ (pushˡ assoc-commute-from))) ⟩
+      (α⇐ ∘ id ⊗₁ α⇒ ∘ α⇒ ∘ (id ⊗₁ σ⇒) ⊗₁ id) ∘ α⇒ ⊗₁ id ∘ α⇐
+    ≈⟨ pushʳ sym-assoc ⟩∘⟨refl ⟩
+      ((α⇐ ∘ id ⊗₁ α⇒ ∘ α⇒) ∘ (id ⊗₁ σ⇒) ⊗₁ id) ∘ α⇒ ⊗₁ id ∘ α⇐
+    ≈⟨ pushˡ (conjugate-from (α ⊗ᵢ idᵢ) α (assoc ○ pentagon)) ⟩∘⟨refl ⟩
+      (α⇒ ∘ α⇐ ⊗₁ id ∘ (id ⊗₁ σ⇒) ⊗₁ id) ∘ α⇒ ⊗₁ id ∘ α⇐
+    ≈˘⟨ pushʳ (pushʳ (pushˡ split₁ˡ)) ⟩
+      α⇒ ∘ α⇐ ⊗₁ id ∘ (id ⊗₁ σ⇒ ∘ α⇒) ⊗₁ id ∘ α⇐
+    ≈˘⟨ refl⟩∘⟨ pushˡ split₁ˡ ⟩
+      α⇒ ∘ (α⇐ ∘ id ⊗₁ σ⇒ ∘ α⇒) ⊗₁ id ∘ α⇐
+    ∎
+
+  swapInner-assoc : ∀ {X₁ Y₁ Z₁ X₂ Y₂ Z₂} →
+                    [ ((X₁ ⊗₀ X₂) ⊗₀ (Y₁ ⊗₀ Y₂)) ⊗₀ (Z₁ ⊗₀ Z₂) ⇒
+                      (X₁ ⊗₀ (Y₁ ⊗₀ Z₁)) ⊗₀ (X₂ ⊗₀ (Y₂ ⊗₀ Z₂)) ]⟨
+                      i⇒ ⊗₁ id    ⇒⟨ ((X₁ ⊗₀ Y₁) ⊗₀ (X₂ ⊗₀ Y₂)) ⊗₀ (Z₁ ⊗₀ Z₂) ⟩
+                      i⇒          ⇒⟨ ((X₁ ⊗₀ Y₁) ⊗₀ Z₁) ⊗₀ ((X₂ ⊗₀ Y₂) ⊗₀ Z₂) ⟩
+                      α⇒ ⊗₁ α⇒
+                    ≈ α⇒          ⇒⟨ (X₁ ⊗₀ X₂) ⊗₀ ((Y₁ ⊗₀ Y₂) ⊗₀ (Z₁ ⊗₀ Z₂)) ⟩
+                      id ⊗₁ i⇒    ⇒⟨ (X₁ ⊗₀ X₂) ⊗₀ ((Y₁ ⊗₀ Z₁) ⊗₀ (Y₂ ⊗₀ Z₂)) ⟩
+                      i⇒
+                    ⟩
+  swapInner-assoc = begin
+      α⇒ ⊗₁ α⇒ ∘ i⇒ ∘ i⇒ ⊗₁ id
+    ≈⟨ (begin
+        α⇒ ⊗₁ α⇒
+      ≈⟨ serialize₂₁ ⟩
+        id ⊗₁ α⇒ ∘ α⇒ ⊗₁ id
+      ≈⟨ refl⟩∘⟨ switch-fromtoˡ α (switch-fromtoˡ (idᵢ ⊗ᵢ α) pentagon) ⟩
+        id ⊗₁ α⇒ ∘ α⇐ ∘ id ⊗₁ α⇐ ∘ α⇒ ∘ α⇒
+      ≈˘⟨ refl⟩∘⟨ refl⟩∘⟨ refl⟩⊗⟨ cancelˡ α.isoʳ ⟩∘⟨refl ⟩
+        id ⊗₁ α⇒ ∘ α⇐ ∘ id ⊗₁ (α⇒ ∘ α⇐ ∘ α⇐) ∘ α⇒ ∘ α⇒
+      ≈˘⟨ refl⟩∘⟨ refl⟩∘⟨ refl⟩⊗⟨ pullʳ inv-pentagon ⟩∘⟨refl ⟩
+        id ⊗₁ α⇒ ∘ α⇐ ∘ id ⊗₁ ((α⇒ ∘ α⇐ ⊗₁ id ∘ α⇐) ∘ id ⊗₁ α⇐) ∘ α⇒ ∘ α⇒
+      ≈⟨ refl⟩∘⟨ refl⟩∘⟨ pushˡ split₂ˡ ⟩
+        id ⊗₁ α⇒ ∘ α⇐ ∘ id ⊗₁ (α⇒ ∘ α⇐ ⊗₁ id ∘ α⇐) ∘
+        id ⊗₁ (id ⊗₁ α⇐) ∘ α⇒ ∘ α⇒
+      ≈˘⟨ refl⟩∘⟨ refl⟩∘⟨ refl⟩∘⟨ extendʳ assoc-commute-from ⟩
+        id ⊗₁ α⇒ ∘ α⇐ ∘ id ⊗₁ (α⇒ ∘ α⇐ ⊗₁ id ∘ α⇐) ∘
+        α⇒ ∘ (id ⊗₁ id) ⊗₁ α⇐ ∘ α⇒
+      ≈⟨ refl⟩∘⟨ refl⟩∘⟨ refl⟩∘⟨ refl⟩∘⟨ ⊗.identity ⟩⊗⟨refl ⟩∘⟨refl ⟩
+        id ⊗₁ α⇒ ∘ α⇐ ∘ id ⊗₁ (α⇒ ∘ α⇐ ⊗₁ id ∘ α⇐) ∘ α⇒ ∘ id ⊗₁ α⇐ ∘ α⇒
+      ∎) ⟩∘⟨refl ⟩
+      (id ⊗₁ α⇒ ∘ α⇐ ∘ id ⊗₁ (α⇒ ∘ α⇐ ⊗₁ id ∘ α⇐) ∘ α⇒ ∘ id ⊗₁ α⇐ ∘ α⇒) ∘
```

Comment:
The proof uses the explicit definition of `swapInner` (or maybe I'm misunderstanding what you mean by "explicit"). The `id`s represent the 'straight wires' in the string diagrams (see picture above). They are introduced and eliminated when one needs to move the other parts (specifically the braidings) around in the string diagrams. Without them, it's impossible to apply e.g. the coherence laws for braidings, and even some of the basic coherence laws for monoidal categories. Because we are working directly with monoidal categories and their maps (not with string diagrams) all this becomes explicit.

The basic operations are illustrated in the `Reasoning` module for monoidal categories. E.g.
https://github.com/agda/agda-categories/blob/650f1bf2ece48b8c6d743cb469ed18b9be218534/src/Categories/Category/Monoidal/Reasoning.agda#L93-L102
Assume I want to apply some coherence law that involves `f ⊗₁ h` in the first diagram there. I cannot do it directly because `f` it is part of a composition (with `f`) under the tensor. So I first need to "split" this composition to separate `f ⊗₁ h` from `g`. This introduces a "straight wire" in the form of an `id`. Later on, I might have to re-merge the two. This is one of the reasons why these proofs are so long. Everything needs to be moved around explicitly.

Reply:
My guess is that there will be combinators akin to the `push` and `pull` ones that make a lot of the `assoc`-shuffling noise get highly muted that will enable the same here. It's up to us to discover what they are.

Reply:
Yes. Though these would be 2d-versions of push and pull. I have no idea what that would look like 🙂

Reply:
I think it'll be fun to figure them out. Subsequently.